src/Server/TCPHandler.cpp

```diff
@@ -452,10 +452,14 @@ void TCPHandler::runImpl()
                             if (isQueryCancelled())
                                 return true;
 
+                            if (client_tcp_protocol_version >= DBMS_MIN_PROTOCOL_VERSION_WITH_TIMEZONE_UPDATES
+                                && client_tcp_protocol_version >= DBMS_MIN_REVISION_WITH_SERVER_TIMEZONE)
+                                sendTimezone();
```

Comment:
This is too late, maybe send timezone in `query_context->setInputInitializer` together with an empty header block?
https://github.com/ClickHouse/ClickHouse/blob/master/src/Server/TCPHandler.cpp#L335

Reply:
Let's try and see 🙃

Reply:
OR you can find out where header block is sent and send the timezone before or right after that.
Like https://github.com/ClickHouse/ClickHouse/blob/master/src/Server/TCPHandler.cpp#L810